desk_sensor_table.py

```diff
@@ -0,0 +1,67 @@
+from config.config import CONFIG
+from desk import Desk
+from sensors.omron_d6t.omron_d6t import OmronD6T
+
+
+class DeskSensorTable:
+    """
+    This class is responsible for mapping desk objects to Sensors, and polling sensors occupancy data
+    """
+
+    # In this mapping, each entry is a tuple containing the desk id, the sensor object, and the respective desk object
+    # The purpose of desk id in the tuple is used for easy lookup of desks
+    # The sensor object is the object responsible for reading sensor data and reporting occupancy status
+    # The desk object is used for performing additional operations on the desk if necessary
+    __desk_mapping = []
+
+    def __init__(self):
+        """
+        Initialize the DeskPinTable object. Register all pin event callbacks
+        """
+        self.__load_desk_dictionary()
+
+    def __load_desk_dictionary(self):
+        """
+        Load the config file and initialize the desk mapping object
+        :return: None
+        """
+
+        # Get all desks config
+        all_desks = CONFIG.desks
+
+        # Get info for each desk
+        for d in all_desks:
+
+            # Initialize sensor
+            sensor_obj = OmronD6T(d.sensor.i2c_bus, d.sensor.i2c_address)
+
+            # Append id, pin, desk object to map
+            self.__desk_mapping.append((d.id, sensor_obj, Desk(d.name, d.id)))
+
+            # Print additional information in debug mode
+            if CONFIG.debug_config.enabled:
```

Comment:
This sort of behavior (conditional logging) could be handled by the logger if we choose to use a logging library.

Reply:
Yep. Ill have to start adding that into our code base